### PR TITLE
[PM-1380] Modify Device Table

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add new keys to the device table that help facilitate it becoming a trusted device and storing the necessary keys to aid in a trusted device being able to decrypt their vault. 

## Code changes

* **src/Api/Controllers/DevicesController.cs:** Introduce endpoint for putting devices keys after creation of a device, so that a device can be trusted.
* **src/Api/Models/Request/DeviceRequestModels.cs:** Add model for request body that forces all new keys to have been encrypted client side.
* **util/Migrator/DbScripts/2023-05-21_00_AddKeysToDevice.sql:** Add migration for the new `Device` columns.
* Add EF Migrations
* Add SQL Project updates

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Api/Models/Request/DeviceRequestModels.cs [hunk 1]
```diff
@@ -47,3 +48,30 @@ public Device ToDevice(Device existingDevice)
         return existingDevice;
     }
 }
+
+public class DeviceKeysRequestModel
+{
+    /// <inheritdoc cref="Device.EncryptedUserKey" />
```
Comment (jlf0dev) [inline src/Api/Models/Request/DeviceRequestModels.cs:54]: Neat!

**Review by jlf0dev — APPROVED** [review]
Everything looks good to me, @JaredSnider-Bitwarden feel free to skim over and make sure I didn't miss anything.

File: src/Core/Entities/Device.cs [hunk 2]
```diff
@@ -17,6 +17,26 @@ public class Device : ITableObject<Guid>
...
+
+    /// <summary>
+    /// Intended to be the public key that was generated for a device upon trust and encrypted. Currenly encrypted using
+    /// a users symmetric key so that when trusted and unlocked a user can decrypt the public key for all their devices.
+    /// This enabled a user to rotate the keys for all of their devices.
+    /// </summary>
+    public string EncryptedPublicKey { get; set; }
+
+    /// <summary>
+    /// Intended to be the public key that was generated for a device upon trust and encrypted. Currenly encrypted with
```
Comment (JaredSnider-Bitwarden) [inline src/Core/Entities/Device.cs:34]: Great documentation overall! Should this be Private key here?

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM! Thank you for these excellent changes!

**Comment by justindbaur** [comment]
Merging to master so QA can fully test login with device.

**Pull request merged**